pysindy/optimizers/sindy_pi.py

```diff
@@ -158,27 +150,25 @@ def _update_parallel_coef_constraints(self, x):
             print("Model ", i)
             xi = cp.Variable(n_features)
             # Note that norm choice below must be convex,
-            # so thresholder must be L1 or L2
-            if (self.thresholder).lower() in ("l1", "weighted_l1"):
-                if self.thresholds is None:
-                    cost = cp.sum_squares(x[:, i] - x @ xi) + self.threshold * cp.norm1(
-                        xi
-                    )
-                else:
-                    cost = cp.sum_squares(x[:, i] - x @ xi) + cp.norm1(
-                        self.thresholds[i, :] @ xi
-                    )
-            if (self.thresholder).lower() in ("l2", "weighted_l2"):
-                if self.thresholds is None:
-                    cost = (
-                        cp.sum_squares(x[:, i] - x @ xi)
-                        + self.threshold * cp.norm2(xi) ** 2
-                    )
-                else:
-                    cost = (
-                        cp.sum_squares(x[:, i] - x @ xi)
-                        + cp.norm2(self.thresholds[i, :] @ xi) ** 2
-                    )
+            # so regularizer must be L1 or L2
+            regularizer = self.regularizer.lower()
+            lam = self.reg_weight
+            if regularizer == "l1":
+                cost = cp.sum_squares(x[:, i] - x @ xi) + cp.sum(
+                    cp.multiply(lam, cp.abs(xi))
+                )
+            elif regularizer == "weighted_l1":
+                cost = cp.sum_squares(x[:, i] - x @ xi) + cp.sum(
+                    cp.multiply(lam[:, i], cp.abs(xi))
+                )
+            elif regularizer == "l2":
+                cost = cp.sum_squares(x[:, i] - x @ xi) + cp.sum(
+                    cp.multiply(lam, xi**2)
+                )
+            elif regularizer == "weighted_l2":
+                cost = cp.sum_squares(x[:, i] - x @ xi) + cp.sum(
+                    cp.multiply(lam[:, i], xi**2)
+                )
```

Comment:
Can we find a way to use `_calculate_penalty`?

Reply:
The logic is slightly different from ConstrainedSR3. I'm not sure if we can combine them.

Reply:
If it doesn't work, override the superclass `_calculate_penalty()`